spec/unit/provider/remote_file/sftp_spec.rb

```diff
@@ -127,7 +127,7 @@
     end
 
     context "and the uri specifies a nested path" do
-      let(:uri) { URI.parse("ftp://conan:cthu1hu@opscode.com/the/whole/path/seattle.txt") }
+      let(:uri) { URI.parse("sftp://conan:cthu1hu@opscode.com/the/whole/path/seattle.txt") }
 
       it "should fetch the file from the correct path" do
         expect(sftp).to receive(:download!).with("the/whole/path/seattle.txt", "/tmp/somedir/remote-file-sftp-backend-spec-test")
```

Comment:
just add a / in front of the/whole/path/seattle.txt.  This is compliant with the sftp url spec and was missed because the original url typo.  Thanks for catching this!